File: src/client/app/transactions/transaction.ts
import { randomId } from '../shared/utils';

export class Transaction {
  id: string = randomId();
  title: string;
  description: string = null;
  date: Date = new Date();
  amount: number;
  expense = true;
  categoryId: string;
  budgetId: string;
  createdBy: string;
}

File: core/src/main/kotlin/com/wbrawner/twigs/model/Transaction.kt
package com.wbrawner.twigs.model

import com.wbrawner.twigs.Identifiable
import com.wbrawner.twigs.randomString
import java.time.Instant

data class Transaction(
    override val id: String = randomString(),
    val title: String,
    val description: String? = null,
    val date: Instant,
    val amount: Long,
    val expense: Boolean,
    val createdBy: String,
    val categoryId: String? = null,
    val budgetId: String
) : Identifiable
